### PR TITLE
feat: 🎸added disable option in menuItem for disabling dropdown option

### PR DESCRIPTION
**Pull request by saadurrehman-selectquote**
feat: 🎸 In this feature we can disable dropdown option

In this feature we can disable SQFormdropdown option specifically by
providing isDisable value

✅ Closes: 118

Screenshot:
image

### CODE REVIEW

**Review by SeanGroff — CHANGES_REQUESTED** [review]
This is not a breaking change. In the commit it says there is a breaking change. 

When there's no breaking change, during the commit wizard choose N or press enter.

**Comment by saadurrehman-selectquote** [comment]
> This is not a breaking change. In the commit it says there is a breaking change.
> 
> When there's no breaking change, during the commit wizard choose N or press enter.

Done

**Comment by SeanGroff** [comment]
> > This is not a breaking change. In the commit it says there is a breaking change.
> 
> > 
> 
> > When there's no breaking change, during the commit wizard choose N or press enter.
> 
> 
> 
> Done

image

The commit still shows a breaking change?

File: src/components/SQForm/SQFormDropdown.js [hunk 1]
```diff
@@ -79,7 +79,11 @@ function SQFormDropdown({
       >
         {options.map(option => {
           return (
-            <MenuItem key={option.value} value={option.value}>
+            <MenuItem
+              key={option.value}
+              disabled={option.isDisabled}
```
Comment (20BBrown14) [inline src/components/SQForm/SQFormDropdown.js:84]: We should add this `isDisabled` key to the `children` propTypes shape as optional

Comment (saadurrehman-selectquote) [reply src/components/SQForm/SQFormDropdown.js:84]: This property will be used if we provide isDisbaled in the options
This is not passing as a prop in SQFormDropdown 
example: const MOCK_STATE_OPTIONS = [
  {label: 'Arizona', value: 'AZ', isDisabled: true},
  {label: 'Kansas', value: 'KS',  isDisabled: true},
];

**Comment by 20BBrown14** [comment]
> The commit still shows a breaking change?

FWIW I think even if the PR description doesn't show the breaking change the commit still has the breaking change and the bot(s) will see the breaking change when it matters. For instance, if you `git log` this commit the breaking change is still there. I think the whole commit will need to be remade.

File: src/components/SQForm/SQFormDropdown.js (hunk 1 above)
Comment (20BBrown14) [reply src/components/SQForm/SQFormDropdown.js:84]: It's being passed as an optional part of the children to SQFormDropdown. SQFormDropdown accepts it's dropdown options as children.

**Comment by saadurrehman-selectquote** [comment]
> > The commit still shows a breaking change?
> 
> FWIW I think even if the PR description doesn't show the breaking change the commit still has the breaking change and the bot(s) will see the breaking change when it matters. For instance, if you `git log` this commit the breaking change is still there. I think the whole commit will need to be remade.

Should i decline and create PR again ?

**Pull request closed without merging**